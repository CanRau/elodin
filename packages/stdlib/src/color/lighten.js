import Color from 'color'

export default function lighten(color, percentage, type = 'rgb') {
  return Color(color)
    .lighten(percentage)
    [type]()
    .string()
}
