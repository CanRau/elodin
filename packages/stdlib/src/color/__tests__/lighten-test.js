import lighten from '../lighten'

describe('Lightening a color', () => {
  it('should return a string with the input format', () => {
    expect(lighten('rgb(100, 100, 100)', 0.5, 'rgb')).toMatchSnapshot()
    expect(lighten('rgba(100, 100, 100, 0.5)', 0.5, 'rgb')).toMatchSnapshot()
  })
})
