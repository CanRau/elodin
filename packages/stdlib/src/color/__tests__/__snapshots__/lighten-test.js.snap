// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightening a color should return a string with the input format 1`] = `"rgb(150, 150, 150)"`;

exports[`Lightening a color should return a string with the input format 2`] = `"rgba(150, 150, 150, 0.5)"`;
